engine/src/em-dc.cpp

```diff
@@ -99,15 +100,39 @@ bool MCEmscriptenHandleMousePress(MCStack *p_stack, uint32_t p_time, uint32_t p_
 	if (MCnoui) return false;
 	
 	MCScreenDC *t_dc = static_cast<MCScreenDC *>(MCscreen);
+	t_dc->update_mouse_press_state(p_state, p_button);
 	
-	t_dc->handle_mouse_press(p_stack, p_time, p_modifiers, p_state, p_button);
+	MCEventQueuePostMousePress(p_stack, p_time, p_modifiers, p_state, p_button);
+	
+	return true;
+}
+
+static inline MCPoint MCEmscriptenWindowToGlobalLoc(MCStack *p_stack, const MCPoint &p_loc)
+{
+	MCRectangle t_window_rect = p_stack->view_getrect();
+	return MCPointOffset(p_loc, t_window_rect.x, t_window_rect.y);
+}
+
+MC_DLLEXPORT_DEF
+bool MCEmscriptenHandleMousePosition(MCStack *p_stack, uint32_t p_time, uint32_t p_modifiers, int32_t p_x, int32_t p_y)
+{
+	if (MCnoui) return false;
+	
+	MCScreenDC *t_dc = static_cast<MCScreenDC *>(MCscreen);
+	
+	MCPoint t_position = MCPointMake(p_x, p_y);
+	if (p_stack)
+		t_position = MCEmscriptenWindowToGlobalLoc(p_stack, t_position);
+	if (t_dc->update_mouse_position(t_position))
+		MCEventQueuePostMousePosition(p_stack, p_time, p_modifiers, p_x, p_y);
 	
 	return true;
 }
 
 MCScreenDC::MCScreenDC()
 	: m_main_window(nil), m_mouse_button_state(0)
 {
+	m_mouse_position = MCPointMake(-1,-1);
```

Comment:
Would `: m_main_window(nil), m_mouse_button_state(0), m_mouse_position({-1, -1})` work here? Just curious as it would be slightly more efficient (not that this class is constructed very often :-D)

Reply:
Something like `m_mouse_position{-1, -1}` should work I think.